### ISSUE TITLE
New rule proposal: TSLint's no-string-literal

### ISSUE THREAD

**Issue by gruckion** [eslint/eslint#11784]
**Please describe what the rule should do:**

Forbids unnecessary string literal property access. Allows obj["prop-erty"] (can’t be a regular property access). Disallows obj["property"] (should be obj.property).

https://palantir.github.io/tslint/rules/no-string-literal/

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

Disallow
```js
obj["property"]
```

Allow
```js
obj["prop-erty"]
obj["property"]
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It was available in TSLint helps keep code clean and readable. In the case of using typescript the problem is If --noImplicitAny is turned off, property access via a string literal will be ‘any’ if the property does not exist. Despite this causing issues in typescript the code style is still generic to both javascript and typescript.

**Are you willing to submit a pull request to implement this rule?**

Maybe

**Comment by platinumazure** [comment]
Hi @gruckion, thanks for the issue.

I think this already exists as [`dot-notation`](https://eslint.org/docs/rules/dot-notation). Please take a look and let me know if you think that rule doesn't fit your needs. Thanks!

**Comment by gruckion** [comment]
Thank you, I will check out this suggestion.

**Comment by g-plane** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
